x/supplier/client/cli/stake_options_reader.go

```
@@ -0,0 +1,157 @@
+	ServiceId string            `yaml:"service_id"`
+	Endpoints []ServiceEndpoint `yaml:"endpoints"`
+}
+
+// ServiceEndpoint is the structure describing a single service endpoint in the stake config file
+type ServiceEndpoint struct {
+	Url     string            `yaml:"url"`
+	RPCType string            `yaml:"rpc_type"`
+	Config  map[string]string `yaml:"config"`
+}
+
+// parseStakeConfig parses the stake config file into a parsedStakeConfig
+func parseStakeConfig(configContent []byte) (*ParsedStakeConfig, error) {
+	var (
+		stakeConfig *StakeConfig
+		err         error
+	)
+
+	// Unmarshal the stake config file into a stakeConfig
+	if err = yaml.Unmarshal(configContent, &stakeConfig); err != nil {
+		return nil, err
+	}
+
+	// Prepare the parsedStakeConfig
+	parsedStakeConfig := &ParsedStakeConfig{}
+
+	// Parse the stake amount and assign it to the parsedStakeConfig
+	parsedStakeConfig.stake, err = sdk.ParseCoinNormalized(stakeConfig.Stake)
+	if err != nil {
+		return nil, err
+	}
+
+	// Prepare the services slice
+	var services []*sharedtypes.SupplierServiceConfig
+
+	// Populate the services slice
+	for _, svc := range stakeConfig.Services {
+		// Validate the serviceId
+		// TODO_TECH_DEBT: This should be validated against some governance state
+		// defining the network's supported services
+		if svc.ServiceId == "" {
+			return nil, fmt.Errorf("invalid serviceId")
+		}
+
+		// Create a supplied service config with the serviceId
+		service := &sharedtypes.SupplierServiceConfig{
+			Service:   &sharedtypes.Service{Id: svc.ServiceId},
+			Endpoints: []*sharedtypes.SupplierEndpoint{},
+		}
+
+		// Iterate over the service endpoints and add their parsed representation to the supplied service config
+		for _, endpoint := range svc.Endpoints {
+			parsedEndpointEntry, err := parseEndpointEntry(endpoint)
+			if err != nil {
+				return nil, err
+			}
+			service.Endpoints = append(service.Endpoints, parsedEndpointEntry)
+		}
+		services = append(services, service)
```

We should throw an error if `len(endpoints) == 0`